recipes/criterion/all/conandata.yml

```diff
@@ -0,0 +1,12 @@
+sources:
+  "2.4.1":
+    "source":
+      url: "https://github.com/Snaipe/Criterion/archive/refs/tags/v2.4.1.zip"
+      sha256: "033c3ac4d9b7cc1bc64ca0cfa352a90b7f33aa8ae077193944457f15eda691be"
+    "dependencies":
```

Comment:
These should be separate recipes :) Each in their own PR, it's exceptionally rare to have things nested but let me know if I missed something

Reply:
Ok, this is a very interesting one. I will start saying Criterion is not mine, I just use it a lot :smile: now, I have to explain a few things about Criterion:

- Their build process is heavily simplified by their usage of wraps in Meson (see the `subprojects` directory in their GH repo)
- I actually can replace some of their dependencies to avoid using those from wraps, that works nicely
- The case with `klib` and `debugbreak` is a good one. In Criterion's repository, they are git submodules, not even dependencies, the creator just uses _specific versions_ of their source code to be included in Criterion, this is why they are not "dependencies" as with those present in wraps... I called them dependencies because some stuff, but in reality they are not :(
- I can't use them as dependencies, again, they are not really dependencies more than pieces of code used in the project (AFAIK)

One option would be treat them more like "patches" that you apply to the source code, but that means I cannot use the simple `apply_conandata_patches` or `patch` because I will need to convert the source code of those two repositories into patches and host them somewhere. If this sounds ok for you (hosting the patch in my personal repo) and treat those as patches so I can use existing patching support, I totally can work on that but I ignore the policy of that in Conan Center :(

Reply:
Ok, I solved the problem for this version, no more hacks regarding the dependencies. I created a patch to solve the problem, in fact, already had a small conversation with the authors of Criterion and they will include my patch with their upcoming version 2.4.2.

This should solve the problem with klib and debugbreak.